paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/instrumentation/ResourcesCompatTransform.kt

```diff
@@ -0,0 +1,66 @@
+package app.cash.paparazzi.gradle.instrumentation
+
+import org.objectweb.asm.ClassVisitor
+import org.objectweb.asm.MethodVisitor
+
+/**
+ * [ClassVisitor] that fixes a hardcoded path in ResourcesCompat.loadFont.
+ * In this method, there is a check that font files have a path that starts with res/.
+ * This is not the case in Studio. This replaces the check with one that verifies that the path contains res/.
+ *
+ * Inspired by ASM fix:
+ * https://cs.android.com/android-studio/platform/tools/adt/idea/+/mirror-goog-studio-main:rendering/src/com/android/tools/rendering/classloading/ResourcesCompatTransform.kt;drc=5bb41b6d5e519c891a4cd6149234138faa28e1af
+ */
+internal class ResourcesCompatTransform(api: Int, delegate: ClassVisitor) : ClassVisitor(api, delegate) {
+  private var isResourcesCompatClass: Boolean = false
+
+  override fun visit(
+    version: Int,
+    access: Int,
+    name: String,
+    signature: String?,
+    superName: String?,
+    interfaces: Array<String>?
+  ) {
+    isResourcesCompatClass = name == RESOURCES_COMPAT_CLASS_NAME
+    super.visit(version, access, name, signature, superName, interfaces)
+  }
+
+  override fun visitMethod(
+    access: Int,
+    name: String,
+    descriptor: String,
+    signature: String?,
+    exceptions: Array<String>?
+  ): MethodVisitor {
+    val methodVisitor = super.visitMethod(access, name, descriptor, signature, exceptions)
+    if (isResourcesCompatClass && name == LOAD_FONT_METHOD_NAME && descriptor == LOAD_FONT_METHOD_DESCRIPTOR) {
+      return LoadFontVisitor(api, methodVisitor)
+    }
+    return methodVisitor
+  }
+
+  private class LoadFontVisitor(api: Int, delegate: MethodVisitor) : MethodVisitor(api, delegate) {
+
+    override fun visitMethodInsn(
+      opcode: Int,
+      owner: String,
+      name: String,
+      descriptor: String,
+      isInterface: Boolean
+    ) {
+      return if ("startsWith" == name) {
```

Comment:
nit: expression function

Reply:
actually, this method returns Unit, so an alternative might be to remove the explicit return?